File: src/features/typing/data/java/merge_intervals.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> a[0] - b[0]);
        List<int[]> merged = new ArrayList<>();
        for (int[] interval : intervals) {
            int[] last = merged.isEmpty() ? null : merged.get(merged.size() - 1);
            if (last == null || last[1] < interval[0]) {
                merged.add(interval);
            } else {
                last[1] = Math.max(last[1], interval[1]);
            }
        }
        return merged.toArray(new int[0][]);
    }
}

File: src/features/typing/data/dart/merge_intervals.dart
class Solution {
  List<List<int>> merge(List<List<int>> intervals) {
    intervals.sort((a, b) => a[0] - b[0]);
    final result = <List<int>>[];
    for (final interval in intervals) {
      if (result.isEmpty || result.last[1] < interval[0]) {
        result.add(interval);
      } else if (interval[1] > result.last[1]) {
        result.last[1] = interval[1];
      }
    }
    return result;
  }
}
